clients/cobol-lsp-vscode-extension/src/task/CompileTaskProvider.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import {taskType} from "../extension";
+import { CustomBuildTaskTerminal } from "./CustomBuildTaskTerminal";
+
+const TASK_DETAIL = "compile with IBM Enterprise COBOL for z/OS";
+export class CompileTaskProvider implements vscode.TaskProvider {
+    public static taskSource = "cobol";
+    public static problemMatcher = "$enterprise-cobol";
+    public static taskProviderName = "compile";
+    private cobolCompileTasks?: Thenable<vscode.Task[]> = undefined;
+
+    public resolveTask(task: vscode.Task, _token: vscode.CancellationToken): vscode.ProviderResult<vscode.Task> {
+        const definition: CobolCompileJobDefinition = <any> task.definition;
```

Comment:
This is a hack. We should use

```js
export class CobolCompileTask extends vscode.Task {
    definition: CobolCompileJobDefinition;
}
```
ad replace the occurrences of `vscode.Task` with `CobolCompileTask` and occurences of `vscode.TaskProvider` with `vscode.TaskProvider<CobolCompileTask>`.

Reply:
awesome! Thanks. Updated as per suggestion